MSAL/src/native_auth/network/MSALNativeAuthCustomErrorSerializer.swift

```diff
@@ -26,9 +26,12 @@ import Foundation
 
 @_implementationOnly import MSAL_Private
 
-final class MSALNativeAuthCustomErrorSerializer<T: Decodable & Error>: NSObject, MSIDResponseSerialization {
+// swiftlint:disable:next line_length
+final class MSALNativeAuthCustomErrorSerializer<T: Decodable & Error & MSALNativeAuthResponseCorrelatable>: NSObject, MSIDResponseSerialization {
     func responseObject(for httpResponse: HTTPURLResponse?, data: Data?, context: MSIDRequestContext?) throws -> Any {
-        let customError = try JSONDecoder().decode(T.self, from: data ?? Data())
+        var customError = try JSONDecoder().decode(T.self, from: data ?? Data())
+        customError.correlationId = customError.retrieveCorrelationIdFromHeaders(from: httpResponse) // DJB: Update the context at this level?
```

Comment:
Should the `// DJB` comment still be here?

Reply:
Done